src/diet.ts

```diff
@@ -589,31 +624,10 @@ export function potionMenu(
     ? potion($item`Boris's bread`, { price: 2 * ingredientCost($item`Yeast of Boris`) })
     : [];
 
-  // Replace string with BooleanProperty later
-  const ofLegendPotion = (item: Item, prefName: string) => {
-    if (get(prefName, true)) return [];
-
-    const recipes = [
-      item,
-      ...$items`roasted vegetable of Jarlsberg, Pete's rich ricotta, Boris's bread`,
-    ].map((i) => toInt(i));
-
-    if (recipes.some((id) => get(`unknownRecipe${id}`, true))) return [];
-
-    return limitedPotion(item, 1, {
-      price:
-        2 *
-        sum($items`Vegetable of Jarlsberg, St. Sneaky Pete's Whey, Yeast of Boris`, ingredientCost),
-    });
-  };
-
-  const ofLegendMenuItems = globalOptions.ascend
-    ? [
-        ...ofLegendPotion($item`Calzone of Legend`, "calzoneOfLegendEaten"),
-        ...ofLegendPotion($item`Pizza of Legend`, "pizzaOfLegendEaten"),
-        ...ofLegendPotion($item`Deep Dish of Legend`, "deepDishOfLegendEaten"),
-      ]
-    : [];
+  const deepDish = legendaryPizzaToMenu(
+    [{ item: $item`Deep Dish of Legend`, pref: "deepDishOfLegendEaten" }],
+    (out: { item: Item; price: number }) => limitedPotion(out.item, 1, { price: out.price })[0]
```

Comment:
will this ever be nullish? the `[0]` has me 👀ing

Reply:
No, but it's very annoying since it could fall apart in the future.